src/rt-app_parse_config.c

```diff
@@ -811,7 +811,7 @@ parse_thread_data(char *name, struct json_object *obj, int index,
 		data->phases = malloc(sizeof(phase_data_t) * data->nphases);
 		parse_thread_phase_data(obj,  &data->phases[0], opts, (long)data);
 		/* Get loop number */
-		data->loop = -1;
+		data->loop = get_int_value_from(obj, "loop", TRUE, -1);
 	}
```

Comment:
Originally it was: data->loop = 1  instead of -1
and everything was working fine when phase was merged in the thread
But viresh raises a problem which i had reproduced but can't reproduce now